File: backend/src/main/java/com/clothingstore/model/purchase/Purchase.java
package com.clothingstore.model.purchase;

import com.clothingstore.model.ShippingInfo;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@NoArgsConstructor
@Getter
@Setter
public class Purchase {

    private String shippingMethod;

    private String paymentMethod;

    private ShippingInfo shippingInfo;

    private List<PurchaseContent> orderContentList;
}

File: frontend/src/app/models/purchase.ts
import { PurchaseContent } from "./purchase-content";
import { ShippingInfo } from "./shipping-info";

export class Purchase {
  shippingMethod : string;
  paymentMethod : string;
  shippingInfo : ShippingInfo;
  orderContentList : Array<PurchaseContent>;

  constructor(shippingMethod : string, paymentMethod : string, shippingInfo : ShippingInfo,
              orderContentList : Array<PurchaseContent>) {
    this.shippingMethod = shippingMethod;
    this.paymentMethod = paymentMethod;
    this.shippingInfo = shippingInfo;
    this.orderContentList = orderContentList;
  }
}
